src/functionsDeployHelper.ts

```diff
@@ -161,10 +184,124 @@ export function getFunctionsInfo(parsedTriggers: CloudFunctionTrigger[], project
+ * @param filters The filters, passed in by the user via  `--only functions:`
+ */
+export function createDeploymentPlan(
+  functionsInLocalSource: RegionMap,
+  existingFunctions: CloudFunctionTrigger[],
+  filters: string[][]
+): DeploymentPlan {
+  const deployment: DeploymentPlan = {
+    regionalDeployments: [],
+    functionsToDelete: [],
+    schedulesToDelete: [],
+  };
+  for (const region of Object.keys(functionsInLocalSource)) {
+    const regionalDeployment: RegionalDeployment = {
+      region,
+      functionsToCreate: [],
+      functionsToUpdate: [],
+      schedulesToCreateOrUpdate: [],
+    };
+    const localFunctionsInRegion = functionsInLocalSource[region];
+    for (const fn of localFunctionsInRegion) {
+      // Check if this function matches the --only filters
+      if (functionMatchesAnyGroup(fn.name, filters)) {
+        // Check if this local function has the same name as an exisiting one.
+        const matchingExistingFunction = _.find(existingFunctions, (exFn) => {
+          return exFn.name === fn.name;
+        });
+        // Check if the matching exisitng function is scheduled
+        const isMatchingExisitingFnScheduled =
+          _.get(matchingExistingFunction, "labels.deployment-scheduled") === "true";
+        // Check if the local function is a scheduled function
+        const isScheduled = _.has(fn, "schedule");
+
+        if (!matchingExistingFunction) {
+          regionalDeployment.functionsToCreate.push(fn);
+        } else {
+          regionalDeployment.functionsToUpdate.push(fn);
+          _.remove(existingFunctions, (exFn: CloudFunctionTrigger) => {
+            return exFn.name === fn.name;
+          });
+        }
+        // Check for schedules.
+        if (isScheduled) {
+          // If the local function is scheduled, create or update a schedule.
+          regionalDeployment.schedulesToCreateOrUpdate.push(fn);
+        } else if (!isScheduled && isMatchingExisitingFnScheduled) {
+          // If the local function isn't scheduled but the existing one is, delete the schedule.
+          deployment.schedulesToDelete.push(matchingExistingFunction!.name);
+        }
+      }
+    }
+    deployment.regionalDeployments.push(regionalDeployment);
+  }
+
+  // Delete any remaining existing functions that:
+  // 1 - Have the deployment-tool: 'firebase-cli' label and
+  // 2 - Match the --only filters, if any are provided.
+  const functionsToDelete = _.chain(existingFunctions)
+    .filter((fn) => {
+      return deploymentTool.check(fn.labels);
+    })
+    .filter((fn) => {
+      return filters.length ? functionMatchesAnyGroup(fn.name, filters) : true;
+    })
+    .value();
+  deployment.functionsToDelete = _.map(functionsToDelete, (fn) => {
+    return fn.name;
+  });
+  // Also delete any schedules for functions that we are deleting.
+  _.forEach(functionsToDelete, (fn) => {
+    if (_.get(fn, "labels.deployment-scheduled") === "true") {
```

Comment:
I _think_ you could do `fn.labels?.["deployment-scheduled"]  === "true"`

(the last thing is `["thing"]` because of the kabab-casing)

Reply:
I believe you're right - plus, its only written this way because I copied this over from old functions code that was written before optional chaining existed. I definitely prefer using the native option, so I'll switch this over.